ci-operator/step-registry/mco/conf/day2/pintargetrelease/mco-conf-day2-pintargetrelease-commands.sh

```diff
@@ -0,0 +1,135 @@
+if [[ -f "${SHARED_DIR}/upgrade-edge" ]]; then
+    echo "ERROR: Serial upgrades are not supported!!"
+    exit 255
+fi
+
+
+function debug() {
+    MCP="${1}"
+
+    oc get mcp -o yaml "${MCP}"
+    oc get pinnedimageset -o yaml "99-${MACHINE_CONFIG_POOL}-pinned-release"
+    for node in $(oc get nodes -l node-role.kubernetes.io/"${MCP}" -o name)
+    do
+       n=${node/node\//}
+       echo "$n"
+       oc get machineconfignode "$n" -ojsonpath='{.status.conditions[?(@.type=="PinnedImageSetsDegraded")]}' | jq
+       oc get machineconfignode "$n" -ojsonpath='{.status.conditions[?(@.type=="PinnedImageSetsProgressing")]}' | jq
+    done
+}
+
+function wait_for_mcp_to_start_pinning_images() {
+    MCP="${1}"
+    TIME=5
+    TOTAL=0
+    TIME_OUT=180
+
+    echo "Waiting for MCP ${MCP} to start pinning the images"
+
+    while [ "$(oc get mcp worker -ojsonpath='{.status.poolSynchronizersStatus[?(@.poolSynchronizerType=="PinnedImageSets")].updatedMachineCount}')"  == \
+    	"$(oc get mcp worker -ojsonpath='{.status.machineCount}')" ]; do
+
+	TOTAL=$((TIME + TOTAL))
+
+	if (( TOTAL > TIME_OUT  )); then
+            echo "MCP ${MCP} is taking to long to start pinning images"
+            # Print debug info
+            debug "${MCP}"
+        fi
+
+	echo "Waiting ${TIME} seconds..."
+        sleep $TIME
+    done
+
+    echo "Pool $MCP has started to pin the images" 
+}
+
+function wait_for_mcp_to_finish_pinning_images() {
+    MCP="${1}"
+    TIME=60
+    TOTAL=0
+    TIME_OUT=3000
+
+    echo "Waiting for MCP ${MCP} to finish"
+
+    while [ "$(oc get mcp worker -ojsonpath='{.status.poolSynchronizersStatus[?(@.poolSynchronizerType=="PinnedImageSets")].updatedMachineCount}')"  != \
+    	"$(oc get mcp worker -ojsonpath='{.status.machineCount}')" ]; do
+
+	TOTAL=$((TIME + TOTAL))
+
+	if (( TOTAL > TIME_OUT )); then
+            echo "MCP ${MCP} is taking to long to pin the images"
+            # Print debug info
+            debug "${MCP}"
+        fi
+
+	echo "Waiting ${TIME} seconds..."
+        sleep $TIME
+    done
+
+    echo "Pool $MCP has finished" 
+}
+
+
+function get_pinned_image_file_name() {
+    MCP="${1}"
+    echo "pinnedimageset-${MACHINE_CONFIG_POOL}.yaml"
+}
+
+# Writing the PinnedImageSet files
+for MACHINE_CONFIG_POOL in ${MCO_CONF_DAY2_PINTARGETRELEASE_MCPS}; do
+    file=$(get_pinned_image_file_name "${MACHINE_CONFIG_POOL}")
+    echo "Rendering PinnedImageSet for pool ${MACHINE_CONFIG_POOL}"
+
+
+    cat << EOF >> "${file}"
+apiVersion: machineconfiguration.openshift.io/v1alpha1
+kind: PinnedImageSet
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-${MACHINE_CONFIG_POOL}-pinned-release
+spec:
+  pinnedImages:
+$(oc adm release info "${TARGET}" -a "${CLUSTER_PROFILE_DIR}"/pull-secret -o pullspec | awk '{print "   - name: " $1}')
+EOF
+
+# The the release image itself
+echo "   - name: $TARGET_DIGEST" >> "${file}"
```

Comment:
@sergiordlr nice-to-have: put the file copy to `${ARTIFACT_DIR}`

Reply:
Done